terraform/modules/fargate-console/README.md

```diff
@@ -0,0 +1,46 @@
+# Fargate Console
+
+This module provides a way to start a bash session in a running ECS container,
+with the required application code, secrets, and environment variables for
+developers to perform interactive operational and investigative tasks.
+
+Use cases includes:
+
+* Running a Rails console in an AWS environment
+* Running a database console
+* Running a web server without having traffic routed to that server.
+
+See proposal 003 in govuk-replatforming-discovery-2020 for further details
+of the tooling required to make this a good developer experience.
+
+## How it works
+
+Using a GDS CLI command such as `gds govuk console --env test --app frontend`
+people will start a bash session in a running container in ECS.
+
+Behind the scenes this command will start up a container like so:
+
+```shell
+aws ecs run-task --cluster task_runner --task-definition frontend_console \
+--launch-type FARGATE --count 1 --started-by "Harry Potter" \
+--network-configuration '{
+  "awsvpcConfiguration": {
+    "subnets": ["subnet-6dc4370b", "subnet-463bfd0e", "subnet-bfecd0e4"],
+    "securityGroups": ["sg-0b873470482f6232d"],
+    "assignPublicIp": "DISABLED"
+  }
+}'
+```
+
+To SSH onto the container, we must run `ssh -J <username>@<jumbox-url> root@<container-ip>`.
```

Comment:
The companion PR to this has changed a bit since this was written, notably to use AWS SSM instead of raw SSH. That says:

> The tool will poll AWS ECS until the container has the status `RUNNING`, and
then will run a command such as `aws ssm start-session --target <container>`.

Reply:
Thanks, I've removed that and pointed to the proposal.